astropy/modeling/functional_models.py

```diff
@@ -1782,6 +1798,8 @@ def _wrap_wofz(self, z):
         ):
             return self._last_w
 
+        if isinstance(z, u.Quantity):
+            z = z.to_value(u.dimensionless_unscaled)
         self._last_w = self._faddeeva(z)
         self._last_z = z
```

Comment:
```suggestion
        self._last_z = z.to_value(u.dimensionless_unscaled) if isinstance(z, u.Quantity) else z
        self._last_w = self._faddeeva(self._last_z)
```
even a bit shorter actually, but leaving that up to your taste ;-)